core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1939,7 +1939,15 @@ private[spark] object Utils extends Logging {
    * Indicates whether Spark is currently running unit tests.
    */
   def isTesting: Boolean = {
```

Comment:
I understand that this change tried to be conservative to avoid any breakage but `isTesting` isn't an API (it's `private[spark] object Utils` so it's fine to directly change `def isTesting` to `lazy val isTesting`

Reply:
`spark.testing` configuration is a static configuration that doesn't allow runtime modification. So theoretically it's fine to cache in general.

Reply:
Or can we use `System.getenv()` instead directly?

Reply:
There are a testcases which set the property IS_TESTING (e.g. SparkFunSuite), I do not wanted to break them.